presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -152,15 +167,15 @@ protected void setup(Binder binder)
         ServerConfig serverConfig = buildConfigObject(ServerConfig.class);
         FeaturesConfig featuresConfig = buildConfigObject(FeaturesConfig.class);
 
-        // TODO: this should only be installed if this is a coordinator
-        binder.install(new CoordinatorModule(featuresConfig.isResourceGroupsEnabled()));
-
         if (serverConfig.isCoordinator()) {
-            discoveryBinder(binder).bindHttpAnnouncement("presto-coordinator");
+            install(new CoordinatorModule());
```

Comment:
`binder.install()` doesn't work. That's why we have the `install` method in `AbstractConfigurationAwareModule`.

(The problem with that method is that it takes `ConfigurationAwareModule` rather than `Module`, so you can't use it for everything, and if a module later becomes configuration aware, you have to go and change the call to make it work.)